core/shared/src/main/scala/org/scalacheck/Prop.scala

```diff
@@ -784,7 +787,8 @@ object Prop {
     def shrinker(x: T, r: Result, shrinks: Int, orig: T): Result = {
       val xs = shrink(x)
       val res = r.addArg(Arg(labels,x,shrinks,orig,pp(x),pp(orig)))
-      if(xs.isEmpty) res else getFirstFailure(xs) match {
+      val originalException = Some(r.status).collect { case Exception(e) => e.getClass() }
```

Comment:
Do you want to skip other `Throwables` that are `NonFatal`?

Reply:
No there is no particular reason. I'll match on `NonFatal` instead.